src/TabBar.tsx

```diff
@@ -155,7 +178,41 @@ export default class TabBar<T extends Route> extends React.Component<
   };
 
   private getScrollAmount = (props: Props<T>, state: State, i: number) => {
-    const { layout } = state;
+    const { layout, tabWidths } = state;
+
+    if (props.dynamicWidth) {
+      const targetIndex =
+        i > props.navigationState.index &&
+        i < props.navigationState.routes.length - 1
+          ? Math.ceil(i)
+          : Math.floor(i);
+      let dynamicAmount = tabWidths
+        .slice(0, targetIndex)
+        .reduce(
+          (acc: number, tabWidth: number) => (acc += tabWidth),
+          tabWidths[targetIndex] * 0.5
+        );
+
+      let diff =
+        tabWidths[targetIndex] * 0.5 +
+        tabWidths[props.navigationState.index] * 0.5;
+
+      if (i > props.navigationState.index) {
```

Comment:
I'm a bit confused about why the current navigation state matters when scrolling to an index. This function should scroll to a specific index without caring about current state.

Though this function is triggered when the index changes, so `i` is the same as the current index anyway.